daft/dataframe/dataframe.py

```diff
@@ -2536,6 +2536,36 @@ def intersect(self, other: "DataFrame") -> "DataFrame":
         builder = self._builder.intersect(other._builder)
         return DataFrame(builder)
 
+    @DataframePublicAPI
+    def intersect_all(self, other: "DataFrame") -> "DataFrame":
+        """Returns the intersection of two DataFrames, including duplicates.
+
+        :param other:
+        :return:
+        """
+        builder = self._builder.intersect_all(other._builder)
+        return DataFrame(builder)
+
+    @DataframePublicAPI
+    def except_distinct(self, other: "DataFrame") -> "DataFrame":
+        """
+
+        :param other:
+        :return:
+        """
+        builder = self._builder.except_distinct(other._builder)
+        return DataFrame(builder)
+
+    @DataframePublicAPI
+    def except_all(self, other: "DataFrame") -> "DataFrame":
+        """
+
+        :param other:
+        :return:
+        """
+        builder = self._builder.except_all(other._builder)
+        return DataFrame(builder)
+
```

Comment:
🤔 I'm wondering if it'd be better to have parametrized functions instead. 
```
intersect_all -> intersect(other, all=True)
except_all -> except(other, all=True)
except_distinct -> except(other, distinct=True)
```

I don't have any strong opinions on this, but I'd take a close look at how other dataframe engines expose this. _(spark, polars, datafusion, ibis, ...)_

Reply:
thanks for the thorough analysis! With all this in mind, I'm fine with the current choices, but I'd like to get at least one other reviewer to sign off as well.

Reply:
I think having something like `intersect` and `intersect_all` be separate functions makes sense to me. This is sort of similar to our `count` and `count_distinct` expressions which are also separated. I'm also happy to follow the convention set by Spark